src/tlapm_args.ml

```diff
@@ -134,140 +107,254 @@ let quote_if_needed s =
+  Arg.(value & opt int 1 & info ["toolbox-vsn"] ~docv:"1|2" ~doc)
+
+let line_t =
+  let doc = "Line to prove." in
+  Arg.(value & opt (some int) None & info ["line"] ~docv:"LINENO" ~doc)
+
+let wait_t =
+  let doc = "Wait for $(docv) before printing obligations in progress." in
+  Arg.(value & opt int 3 & info ["wait"] ~docv:"TIME" ~doc)
+
+let stdin_t =
+  let doc = "Read the tla file from stdin instead of file system. \
+             Only applies if single tla file is provided as input." in
+  Arg.(value & flag & info ["stdin"] ~doc)
+
+let prefer_stdlib_t =
+  let doc = "Prefer build-in standard modules if the module search path \
+             contains files with the same names as modules in stdlib." in
+  Arg.(value & flag & info ["prefer-stdlib"] ~doc)
+
+let noproving_t =
+  let doc = "Do not prove, report fingerprinted results only." in
+  Arg.(value & flag & info ["noproving"] ~doc)
+
+let printallobs_t =
+  let doc = "Print obligations in all toolbox messages." in
+  Arg.(value & flag & info ["printallobs"] ~doc)
+
+let safefp_t =
+  let doc = "Check tlapm, zenon, Isabelle versions for fingerprints." in
+  Arg.(value & flag & info ["safefp"] ~doc)
+
+let nofp_t =
+  let doc = "Do not use existing fingerprints, \
+             but overwrite any preexisting fingerprints associated \
+             with the current proof obligations, \
+             using the newly computed fingerprints and results" in
+  Arg.(value & flag & info ["nofp"] ~doc)
+
+let nofpl_t =
+  let doc = "Disable fingerprint use between given lines." in
+  Arg.(value & opt (some (pair int int)) None & info ["nofpl"] ~doc)
+
+let cleanfp_t =
+  let doc = "Erase fingerprint file before starting." in
+  Arg.(value & flag & info ["cleanfp"] ~doc)
+
+let erasefp_t =
+  let doc = "Erase from file FILE all results of backend BACK." in
+  Arg.(value & opt (some (pair file string)) None & info ["erasefp"] ~docv:"FILE,BACK" ~doc)
+
+let printfp_t =
+  let doc = "Print the fingerprints stored in file $(docv) and quit." in
+  Arg.(value & opt (some file) None & info ["printfp"] ~docv:"FILE" ~doc)
+
+let usefp_t =
+  let doc = "Load fingerprints from file $(docv) (save as usual)." in
+  Arg.(value & opt (some file) None & info ["usefp"] ~docv:"FILE" ~doc)
+
+let fpp_t =
+  let doc = "Print the fingerprints of obligations in toolbox messages." in
+  Arg.(value & flag & info ["fpp"] ~doc)
+
+let cache_dir_t =
+  let doc = "Save auxiliary and \
+             temporary files under $(docv). \
              Alternatively, this directory \
              can be defined via the variable \
              `TLAPM_CACHE_DIR` of the runtime \
              environment. The command-line \
              parameter takes precedence over \
              the environment variable. If neither \
-             is specified, the default value is \".tlacache\".";
-  ]
+             is specified, the default value is \".tlacache\"." in
+  Arg.(value & opt (some dir) None & info ["cache-dir"] ~docv:"DIR" ~doc)
```

Comment:
Could make `.tlacache` the default value of this parameter instead of spelling it out in the docs?

Reply:
I don't think this would be correct, since the command-line option overrides the environment variable `TLAPM_CACHE_DIR`, and `.tlacache` is used only when neither is given.